Api/Services/RestaurantService.cs

```diff
@@ -1368,5 +1370,86 @@ public async Task<Result<RestaurantVM>> GetRestaurantByIdAsync(int restaurantId)
 
             return result;
         }
+
+        /// <summary>
+        /// Returns a list of menus of specific restaurant (owner version)
+        /// </summary>
+        /// <param name="id"> Id of the restaurant.</param>
+        /// <returns></returns>
+        public async Task<List<MenuSummaryVM>?> GetMenusOwnerAsync(int id,User user)
+        {
+            var restaurant = await context.Restaurants
+                .Include(r => r.Menus)
+                .Where(i => i.Id == id)
+                .FirstOrDefaultAsync();
+
+            if (restaurant == null)
+            {
+                return null;
+            }
+
+            var result = await accesService.verifyOwnerRole(id, user);
+            if(!result.Value)
+            {
+                return null;
+            }
+
+
+            var menus = restaurant.Menus
+                .Select(menu => new MenuSummaryVM
+                {
+                    MenuId = menu.Id,
+                    Name = menu.Name,
+                    AlternateName = menu.AlternateName,
+                    MenuType = menu.MenuType,
+                    DateFrom = menu.DateFrom,
+                    DateUntil = menu.DateUntil,
+                    Photo = uploadService.GetPathForFileName(menu.PhotoFileName)
+                })
+                .ToList();
+
+            return menus;
+        }
+
+
+        /// <summary>
+        /// Validates and gets menu items from the given restaurant
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="restaurantId"></param>
+        /// <returns>MenuItems</returns>
+        public async Task<Result<List<MenuItemVM>>> GetMenuItemsOwnerAsync(User user, int restaurantId)
+        {
+            var isRestaurantValid = await menuItemsService.ValidateRestaurant(user, restaurantId);
+
+            if (isRestaurantValid.IsError)
+            {
+                return isRestaurantValid.Errors;
+            }
```

Comment:
To już waliduje czy resauracja istnieje i czy użytkownik ma do niej dostęp, możesz ją usunąć lub przenieść do AccessService i używać jej